source/visionEnhancementProviders/_exampleProvider_autoGui.py

```diff
@@ -0,0 +1,190 @@
+				defaultVal=False
+			),
+			driverHandler.NumericDriverSetting(
+				"amountOfZ",  # value stored in matching property name on class
+				"Amount of Z",
+				defaultVal=11
+			),
+			driverHandler.DriverSetting(
+				# options for this come from a property with name generated by
+				# f"available{settingID.capitalize()}s"
+				# Note:
+				#   First letter of Id becomes capital, the rest lowercase.
+				#   the 's' character on the end.
+				# result: 'availableNameofsomethings'
+				"nameOfSomething",  # value stored in matching property name on class
+				"Name of something",
+			)
+		]
+
+	def clearRuntimeSettingAvailability(self):
+		self._availableRuntimeSettings = []
+
+	def addRuntimeSettingsAvailibility(self, settingIDs: List[str]):
+		self._availableRuntimeSettings.extend(settingIDs)
+		# ensure any previously saved settings are loaded from config file:
+		self._initSpecificSettings(self, self._getAvailableRuntimeSettings())
+
+	def _hasFeature(self, settingID: str) -> bool:
+		return settingID in self._availableRuntimeSettings
+
+	def _getAvailableRuntimeSettings(self) -> SupportedSettingType:
+		settings = []
+		if self._hasFeature("runtimeOnlySetting_externalValueLoad"):
+			settings.extend([
+				driverHandler.NumericDriverSetting(
+					"runtimeOnlySetting_externalValueLoad",  # value stored in matching property name on class
+					"Runtime Only amount, external value load",
+					# no GUI default
+				),
+			])
+		if self._hasFeature("runtimeOnlySetting_localDefault"):
+			settings.extend([
+				driverHandler.NumericDriverSetting(
+					"runtimeOnlySetting_localDefault",  # value stored in matching property name on class
+					"Runtime Only amount, local default",
+					defaultVal=50,
+				),
+			])
+		return settings
+
+	def _get_supportedSettings(self) -> SupportedSettingType:
+		settings = []
+		settings.extend(self.preInitSettings)
+		settings.extend(self._getAvailableRuntimeSettings())
+		return settings
+
+
+class AutoGuiTestProvider(vision.providerBase.VisionEnhancementProvider):
+	_settings = AutoGuiTestSettings()
+
+	@classmethod
+	def canStart(cls):
+		return True  # Check any dependencies (Windows version, Hardware access, Installed applications)
+
+	@classmethod
+	def getSettingsPanelClass(cls) -> Optional[Type]:
+		"""Returns the instance to be used in order to construct a settings panel for the provider.
+		@return: Optional[SettingsPanel]
+		@remarks: When None is returned, L{gui.settingsDialogs.VisionProviderSubPanel_Wrapper} is used.
+		"""
+		return None  # No custom GUI
+
+	@classmethod
+	def getSettings(cls) -> AutoGuiTestSettings:
+		return cls._settings
+
+	def __init__(self):
+		super().__init__()
+		self._initRuntimeOnlySettings()
+		self._showCurrentConfig()
+
+	def _initRuntimeOnlySettings(self):
+		""" This method might query another application for its capabilities and initialise these configuration
+			options.
+		"""
+		settings = self.getSettings()
+		settings.addRuntimeSettingsAvailibility([

```

Comment:
I think it would help if this could be abstracted a bit, so that it is easier for custom providers to add runtime only settings without copying much of this code.

How about the following:

1. On VisionEnhancementProviderSettings, add a registerInstance method. When called, the method should store a weakref to the instance of the provider on the instance of VisionEnhancementProviderSettings.
1. The base implementation of VisionEnhancementProvider can call this registerInstance method when initializing. It can optionally unregister itself when calling terminate, though a weakref will deal with unregistration automatically.
1. VisionEnhancementProviderSettings can expose the runtime only settings when getting the supportedSettings property and there is a registered instance. If not, it will only expose the preInit settings.

Reply:
I think I'd prefer not to be prescriptive about this, since we don't have any current inbuilt code depending on runtime settings. The `_exampleProvider_autoGui.py` shows one way you might do this. 

If you are happy to try to get this merged as it is, perhaps you can make a draft PR afterwards to explore these ideas further?

Reply:
Yeah, I"m fine with that, though filing the draft probably won't have that much priority. If there's one or even more ways to do it, that's fine, as long as it is possible.